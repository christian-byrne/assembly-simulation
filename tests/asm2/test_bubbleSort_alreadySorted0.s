.data

COMMA_SEP:
	.asciiz ", "
    
.globl   printInts
printInts:
	.byte 0

.globl   printWords
printWords:
	.byte 0

.globl   bubbleSort
bubbleSort:
	.byte 1

.globl   printInts_howToFindLen
printInts_howToFindLen:
	.half 0

.globl   intsArray
intsArray:
	.word -3000
	.word -1000
	.word 2000
	.word 5000
	.word 7000

.globl   intsArray_END
intsArray_END:
	

.globl   intsArray_len
intsArray_len:
	.word 5

.globl   theString
theString:
	.asciiz "DO NOT TOUCH!"


.globl   kLuanrtCSLik
kLuanrtCSLik:
	.asciiz "\n+-------------------------------+\n|    Testcase Variable Dump:    |\n+-------------------------------+\n\n"

.globl   zDLEOltKKmpF
zDLEOltKKmpF:
	.asciiz "==============================================\nTEST (bubbleSort_alreadySorted0):\n\nprintInts=0\nprintWords=0\nbubbleSort=1\nprintInts_howToFindLen=0\nintsArray=[-3000, -1000, 2000, 5000, 7000]\nintsArray_len=5\ntheString=\"DO NOT TOUCH!\"\n\n==============================================\n\n"


.text

.globl main

main:
	addi $v0, $zero, 4	# print
	la $a0, zDLEOltKKmpF
	syscall

	jal studentMain

.text
	addi $v0, $zero, 4	# print
	la $a0, kLuanrtCSLik
	syscall

	addi $v0, $zero, 11	# print
	addi $a0, $zero, '['
	syscall

	la $t0, intsArray
    
	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4

	add $t0, $zero, $zero   # reset t0
    
	addi $v0, $zero, 11	# print
	addi $a0, $zero, ']'
	syscall
	addi $v0, $zero, 11
	addi $a0, $zero, '\n'
	syscall

	addi $v0, $zero, 4	# print
	la $a0, theString
	syscall

	addi $v0, $zero, 11
	addi $a0, $zero, '\n'
	syscall


	addi $v0, $zero, 10
	syscall
