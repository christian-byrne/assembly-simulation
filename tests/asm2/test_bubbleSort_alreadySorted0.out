==============================================
TEST (bubbleSort_alreadySorted0):

printInts=0
printWords=0
bubbleSort=1
printInts_howToFindLen=0
intsArray=[-3000, -1000, 2000, 5000, 7000]
intsArray_len=5
theString="DO NOT TOUCH!"

==============================================


+-------------------------------+
|    Testcase Variable Dump:    |
+-------------------------------+

[-3000, -1000, 2000, 5000, 7000]
DO NOT TOUCH!
